### ISSUE TITLE
Core/Creature: Some models should be disabled

### ISSUE THREAD

**Issue by offl** [TrinityCore/TrinityCore#25644]
**Description:**

As we all know sometimes creature has model that is never shown in game but still presents in creature_template. To always use correct model we have to change model in script or in `creature` table. But sometimes player transforms to creature with wrong model and in this case we can't change model in script because creature is not actually spawned, player just takes its model.
Example: https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=37097
Sometimes you will be transformed into an imp instead of a box.

**Expected behaviour:**

Some models should be disabled or we can handle this specific case in HandleAuraTransform or in spell script

**Steps to reproduce the problem:**

1. .go x -3299.180176  2923.374756 171.598175 530 1.877884
2. .cast 37097
3. sometimes you will be transformed into an imp

**Branch(es):** probably both
**TC rev. hash/commit:** 05fa1a46b36f81c177a503f7f5de14eb34830208

**Comment by illfated** [comment]
> Sometimes you will be transformed into an 😈 instead of a 🗳️.

:trollface: 😄  😆

**Comment by ratkosrb** [comment]
I would assume that those unused display ids have probability = 0. In classic data, there are some creatures that have a display id that is not actually used in game, and in some cases the display id does not even exist in client, and you can see the probability for them is 0 in the creature query response (wdb data).

**Comment by offl** [comment]
> and in some cases the display id does not even exist in client, and you can see the probability for them is 0 in the creature query response (wdb data)

Hold on, probability in classic? I discovered that some models are used more often than other when I tried to fix https://github.com/TrinityCore/TrinityCore/issues/24822 for 4/4 models but later I was told that probability thing was not implemented even in 335.

**Comment by ratkosrb** [comment]
Yes, creatures have probabilities for every display id assigned to them, and it's a thing in Classic data too. It has most likely always been a thing, but simply not sent to client back in the day. I'm not sure why they added the probabilities to the packet, probably some new feature in latest expansion required the client to know the probabilities, but they haven't bothered to remove that data from the packet in Classic.

You can check out the structure in WPP.
https://github.com/TrinityCore/WowPacketParser/blob/master/WowPacketParserModule.V1_13_2_31446/Parsers/QueryHandler.cs#L72

In my view the `creature_template_model` table format from master should be used in 3.3.5 branch too, so that probability for picking each display id can be implemented.

**Comment by offl** [comment]
Updated, thanks
Seems like classic can show us lots of things that were never sent in 2xx & 3xx

**Comment by offl** [comment]
From what I see if creature has 2 models (invisible and visible), it will use only invisible model, will be not selectable and will have PASSIVE_REACTSTATE difficulty flag, this means CREATURE_FLAG_EXTRA_TRIGGER will be gone one day or will be removed from nearly all creatures
There is a chance that visible model is used for GMs but it can be achieved without flags simply by sending model without probability

**Comment by Shauren** [comment]
Even if that becomes possible, I'm against removing TRIGGER flag because it allows to write more optimized code in place that really needs it (building SMSG_UPDATE_OBJECT)

**Comment by offl** [comment]
I don't really understand what you mean but anyway, we can make it internal² by removing passive reactstate from its effects and handle passive reactstate by PASSIVE_REACTSTATE difficulty flag
Just to handle this part by one flag and not by two
